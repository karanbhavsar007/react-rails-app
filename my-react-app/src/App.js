import React, { Component } from 'react';
import ReadArticles from './components/ReadArticles/ReadArticles';
import CreateArticle from './components/CreateArticle/CreateArticle';
import ReadOneArticle from './components/ReadOneArticle/ReadOneArticle';
import UpdateArticle from './components/UpdateArticle/UpdateArticle';
import DeleteArticle from './components/DeleteArticle/DeleteArticle';

class App extends Component {
  state = {
    currentMode: 'read',
    articleId: null
  }

  changeAppMode = (newMode, articleId) => {
    this.setState({currentMode: newMode});
            if(articleId !== undefined){
              this.setState({articleId: articleId});
            }
  }

  render() {
    let modeComponent = <ReadArticles changeAppMode={this.changeAppMode}/>
    
    switch(this.state.currentMode){
      case 'read':
          break;
      case 'readOne':
          modeComponent = <ReadOneArticle articleId={this.state.articleId} changeAppMode={this.changeAppMode}/>;
          break;
      case 'create':
          modeComponent = <CreateArticle changeAppMode={this.changeAppMode}/>;
          break;
      case 'update':
          modeComponent = <UpdateArticle articleId={this.state.articleId} changeAppMode={this.changeAppMode}/>;
          break;
      case 'delete':
          modeComponent = <DeleteArticle articleId={this.state.articleId} changeAppMode={this.changeAppMode}/>;
          break;
      default:
          break;
    }
    
    return modeComponent;
  }
}

export default App;
