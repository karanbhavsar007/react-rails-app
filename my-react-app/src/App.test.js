import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
//snapshot of App and if code is changed the test will fail notifying what all was changed
it("renders correctly", () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});