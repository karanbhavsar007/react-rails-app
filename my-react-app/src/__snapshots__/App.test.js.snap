// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    aria-label={null}
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Create Artile
  </button>
  <table
    className="table table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Author
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
